components/level/game.tsx

```diff
@@ -526,7 +530,19 @@ export default function Game({
       event.preventDefault();
     }
 
-    handleKeyDown(code);
+    let timesToMove = 1;
+
+    // if code starts with arrow
+    if (shiftKeyDown.current && code.startsWith('Arrow')) {
```

Comment:
This logic should be moved to within the `setGameState` of `handleKeyDown`

Need a condition around this section of code:
```
      if (!makeMove(newGameState, direction, !disableAutoUndo)) {
        return prevGameState;
      }
```
to check if the shift key is pressed, and if so call makeMove in a loop until it fails

Would need to double check that the `makeMove` function doesn't mess up the gameState object when the function returns false (since it updates in-place)

Reply:
So I was doing this first but then realized that undo may not work as expected. I can try it and see but I think that we wouldn't be storing the intermediate state

Reply:
Interesting... I wonder what would be best there. Maybe it's nice if a single undo removes all the moves at once? Since the shift + move is kind of like one "move"

Reply:
i think it is less confusing if you can undo one at a time and not be in a batch...